test-tools/core/src/main/java/org/kie/cloud/tests/core/resources/Project.java

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.cloud.tests.core.resources;
+
+import java.io.File;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.UUID;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import cz.xtf.builder.builders.ImageStreamBuilder;
+import cz.xtf.core.openshift.OpenShift;
+import cz.xtf.core.openshift.OpenShiftBinary;
+import cz.xtf.core.openshift.OpenShifts;
+import cz.xtf.core.waiting.SimpleWaiter;
+import io.fabric8.kubernetes.api.model.KubernetesList;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.openshift.api.model.ImageStream;
+import org.kie.cloud.tests.core.util.ProcessExecutor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.util.stream.Collectors.toList;
+
+public class Project implements AutoCloseable {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Project.class);
+
+    public static final String POD_STATUS_PENDING = "Pending";
+
+    private String projectName;
+    private OpenShift openShift;
+    private OpenShift openShiftAdmin;
+
+    public Project(String projectName) {
```

Comment:
What about making the constructor private, forcing users to use just getOrCreate() to retrieve project?

Reply:
would make sense.